test/validation/api/classification/odp_classification_common.c

```diff
@@ -318,9 +327,24 @@ odp_packet_t create_packet(cls_packet_info_t pkt_info)
 	vlan_hdr_len = pkt_info.vlan ? ODPH_VLANHDR_LEN : 0;
 	vlan_hdr_len = pkt_info.vlan_qinq ? 2 * vlan_hdr_len : vlan_hdr_len;
 	l3_hdr_len = pkt_info.ipv6 ? ODPH_IPV6HDR_LEN : ODPH_IPV4HDR_LEN;
-	l4_hdr_len = pkt_info.udp ? ODPH_UDPHDR_LEN : ODPH_TCPHDR_LEN;
+	l4_hdr_len = pkt_info.l4_type == CLS_PKT_L4_UDP ? ODPH_UDPHDR_LEN : ODPH_TCPHDR_LEN;
```

Comment:
This sets l4_hdr_len to ODPH_TCPHDR_LEN also in case of SCTP. And same for the other protocols added in later patches. Please move this in the switch statement below.

Reply:
Ack, sent v5.